torch/testing/_internal/composite_compliance.py

```diff
@@ -173,6 +192,107 @@ def wrap(e):
         tree_map(check, rs)
         return rs
 
+
+def is_tensorlist(lst):
+    if not isinstance(lst, list) and not isinstance(lst, tuple):
+        return False
+    if len(lst) == 0:
+        return False
+    return isinstance(lst[0], torch.Tensor)
+
+
+def maybe_map(fn, should_map, arg):
+    return fn(arg) if should_map else arg
+
+
+def wrap(arg):
+    if isinstance(arg, torch.Tensor):
+        return CompositeCompliantTensor(arg)
+    if is_tensorlist(arg):
+        return [CompositeCompliantTensor(a) for a in arg]
+    raise RuntimeError("wrap assumes that the input can be wrapped")
+
+
+# Given a list of flat arguments, some of which may be Tensors, return all
+# possible ways some of the arguments could be CompositeCompliantTensors (CCT).
+# For example, given Tensors A, B, C and flat_args = [A, 1, B],
+# We would return the following 4 options:
+# [CCT(A), 1, CCT(B)]
+# [CCT(A), 1, B]
+# [A, 1, CCT(B)]
+# [A, 1, B]
+# NB: Yes, this is exponential. No, we don't care too much because PyTorch ops
```

Comment:
Are these slow enough that we should consider reducing the number of devices we run on or run them less often?

Reply:
They're not that slow (these tests run in under a minute total on my machine) and people will exile individual tests to the slow tests section if they notice they are slow

Reply:
I ran the stats and recorded the actual information (the one minute ballpark I cited before was from memory)

Total runtime is 1:14 on both CPU and CUDA.
CPU-only is: 32s